build_library/prod_image_util.sh

```diff
@@ -101,7 +102,8 @@ create_prod_image() {
           || die_notrace "coreos-au-key is missing the 'official' use flag"
   fi
 
-  tar -cf "${BUILD_DIR}/${image_pkgdb}" -C "${root_fs_dir}" var/cache/edb var/db/pkg
+   tar -cf "${BUILD_DIR}/${image_pkgdb}" -C "${root_fs_dir}" var/cache/edb var/db/pkg
+  mksquashfs "${root_fs_dir}" "${BUILD_DIR}/configroot" "${BUILD_DIR}"/flatcar_production_image_sysext.sqsh
```

Comment:
The difference of this image and the final image is that the final image has some folders removed and other postprocessing. Instead of archiving one more big artifact, could we rather mount the built `/usr` partition as first lowerdir and as second lowerdir the slim squashfs with the `var/cache/edb` and `var/db/pkg` folders which we could create here? The main difference would be to have either the tar or the squashfs. (You could maybe even not create the squashfs and rather use the existing tar by extracting to a temporary lowerdir).

Reply:
While we would save a few bytes with an incremental approach it would make the whole system more feeble - particularly if we consider making the squashfs a new build artifact to be published for general use (e.g. enabling downstream users to build custom sysexts). I actually like the idea of having a single self-contained squashfs; it makes the approach more generally usable.

With an incremental approach I see 2 problems we wouldn't have with a self-contained image:
1. the squashfs would require the "right" base image to work. We would need to introduce a versioning mechanism to ensure `build_sysext`, e,g, when used on a published release instead of an intermediate step at build time, uses the correct base. A self-contained image on the other hand would ship everything in the right versions - and it would even include a version file itself which `build_sysext` could optionally use to pin a sysext to a Flatcar release.
2. we would need to keep this `tar` line up to date to capture the full state of the OS image before clean-up, introducing additional (manual) maintenance burden. Even today that line doesn't capture all of the state - see the [rm lines right after tar](https://github.com/flatcar/scripts/blob/main/build_library/prod_image_util.sh#L106-L111). This may have subtle side effects in the future depending on where dependencies keep meta-information required for packages at installation time.

Reply:
The usage would be from the SDK and there it already needs to match the right version, or?
If possible, we could reuse the `/usr` partition artifact as first lowerdir: https://stable.release.flatcar-linux.net/amd64-usr/current/flatcar_production_update.bin.bz2 but when it's too complicated, I'm also fine.

Reply:
We can also check that the Flatcar version matches, it has to be the same anyway and this will prevent wrong usage.